zipline/algorithm.py

```diff
@@ -845,6 +858,109 @@ def _create_daily_stats(self, perfs):
 
+        sim_params = self.sim_params
+        data_portal = self.data_portal
+        perf_tracker = self.perf_tracker
+        asset_finder = data_portal.asset_finder
+        lookback_days = zp.DEFAULT_CVAR_LOOKBACK_DAYS
+
+        # Create a data frame of asset weights on each day of the simulation.
+        # If an asset was not held on a given date, it is assigned a weight of
+        # zero.
+        weights = pd.DataFrame(
+            perf_tracker.cumulative_risk_metrics.position_weights.tolist(),
+            index=daily_dts.normalize(),
+        ).fillna(0)
+
+        # For each day of the simulation, check to see if any futures were
+        # held. If so, convert it to a continuous future and assign its weight
+        # on that day to the continuous future. Once that is done, drop the
+        # futures contracts from the weights dataframe as we only want to look
+        # at the continuous futures when doing a history call later on.
+        futures_held = filter(
+            lambda asset: isinstance(asset, Future),
+            weights.columns,
+        )
+        for day in weights.index:
+            for asset in futures_held:
+                if weights.loc[day, asset] != 0:
+                    cf = zp.assets_for_history_call(asset, asset_finder, day)
+                    if cf not in weights:
+                        weights[cf] = 0
+                    weights.loc[day, cf] = weights.loc[day, asset]
+        weights.drop(futures_held, axis=1, inplace=True)
+        assets = weights.columns.tolist()
+
+        benchmark = self._create_benchmark_source().benchmark_asset
+        if benchmark is not None:
+            assets.append(benchmark)
+
+        # If we are near the start date of our data, just use the data
+        # available. Otherwise, use the full default number of lookback days.
+        days_before_start = min(
+            self.trading_calendar.session_distance(
+                data_portal._first_available_session,
+                sim_params.start_session,
+            ),
+            lookback_days,
+        )
+
+        # Get returns values for all assets for the entirety of the simulation.
+        prices = data_portal.get_history_window(
+           assets=assets,
+           end_dt=sim_params.end_session,
+           bar_count=len(sim_params.sessions) + days_before_start,
+           frequency='1d',
+           field='price',
+           data_frequency='daily',
+        )
+        asset_returns = prices.pct_change().iloc[1:]
+
+        # Any assets that came into existence after the start date of the
+        # simulation have their missing returns values proxied with the
+        # benchmark's returns values.
+        if benchmark is not None:
+            for column in asset_returns:
+                asset_returns[column].fillna(
+                    asset_returns[benchmark], inplace=True,
+                )
+            asset_returns.drop(benchmark, axis=1, inplace=True)
```

Comment:
1) whats the right benchmark for futures.
2) Is there a better pandas operation for filling the nans here